framework/db/BaseActiveRecord.php

```diff
@@ -410,6 +411,58 @@ public function hasMany($class, $link)
         return $this->createRelationQuery($class, $link, true);
     }
 
+    /**
+     * Declares a relation from the another class relation definition use `via()`.
+     *
+     * For example, to declare the `items` relation for `Order` class, we can write
+     * the following code in the `Order` class:
+     * ```php
+     * public function getItems() : ItemQuery
+     * {
+     *     return $this->viaRelation('item', 'orderItems');
+     * }
+     * public function getOrderItems()
+     * {
+     *     return $this->hasMany(OrderItem::className(), ['order_id' => 'id']);
+     * }
+     * ```
+     *
+     * and we can write following code in the `OrderItem` class:
+     * ```php
+     * public function getItem()
+     * {
+     *     return $this->hasOne(Item::className(), ['id' => 'item_id']);
+     * }
+     * ```
+     *
+     * Can be used for create a chain of relations, i.e. for `Order` class:
+     * ```php
+     * public function getItemsCategory() : CategoryQuery
+     * {
+     *     return $this->viaRelation('category', 'items');
+     * }
+     * ```
+     *
+     * @param string $relationName relation name in target (case sensitive, without 'get')
+     * @param string $via relation name in called (self) class (case sensitive, without 'get')
+     * @param callable $callable a PHP callback for customizing the relation associated with the junction ActiveQuery in `via`.
+     * Its signature should be `function($query)`, where `$query` is the query to be customized.
+     * @return ActiveQuery ActiveQuery type of end-side model
+     * @since 2.0.14
+     * @see hasOne()
+     * @see hasMany()
+     */
+    public function viaRelation($relationName, $via, callable $callable = null)
+    {
+        $relation = $this->getRelation($via);
+        $target = Yii::createObject($relation->modelClass);
```

Comment:
I forgot for new feature `::instance()`  
(PR is older)  
In this place is correct to use it?

Reply:
Yes.